Phase4/Project/Ranker.cs

```diff
@@ -0,0 +1,41 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Project
+{
+    public class Ranker
+    {
+        private readonly Student[] _students;
+        private readonly StudentScore[] _scores;
+
+        public Ranker(Student[] students, StudentScore[] scores)
+        {
+            _students = students;
+            _scores = scores;
+            CalculateAverages();
```

Comment:
https://softwareengineering.stackexchange.com/questions/208828/passing-an-object-into-a-method-which-changes-the-object-is-it-a-common-anti